contracts/src/ethereum/TokenBridge.sol

```diff
@@ -127,15 +144,15 @@ contract TokenBridge is
     }
 
     function deposit(address l1Token_, uint256 l2Recipient, uint256 amount) external {
-        IERC20 l1Token = IERC20(l1Token_);
+        IStaticATokenLM l1Token = IStaticATokenLM(l1Token_);
```

Comment:
shouldn't we have a modifier here to check whether the l1 token is approved by the bridge or not?

Reply:
good point!